draft-ietf-tls-tls13.md

```diff
@@ -4375,6 +4375,109 @@ currently deployed and properly configured TLS implementations.
 %%### Ticket Establishment
 %%### Updating Keys
 
+# State Machine
+
+This section provides a summary of the legal state transitions for the
+client and server handshakes. State names have no formal meaning but
+are provided in all capitals (e.g., START) for ease of comprehension.
+Messages which are sent only sometimes are indicated in [].
+
+## Client
+
+~~~~
+                +-----> START
+           Recv |        |
+            HRR |        | Send CH
+                |        v
+       +->      +---- WAIT_SH            
+       |                 |                 
+       |                 | Recv SH         
+       |                 |                 
+       |                 V                 
+       |              WAIT_EE            
+       |                 |                 
+       |                 | Recv EE         
+       |                 |                 
+       |          +------+------+          
+       |          |             |              
+       |   SH+PSK |             | SH
+       |          |             v
+       |          |         WAIT_CERT_CR
+   Can |          |          |       |
+  Send |          |          |       |
+ Early |          |     Recv |       | Recv CR
+  Data |          |     Cert |       |
+  Here |          |          |       v
```

Comment:
I'm not sure that you need to highlight the part where early data can be sent.  It's after the first ClientHello, up to the second flight from the client (which might be another ClientHello).  Easier to describe in words and you can reclaim some horizontal space for David's suggestion.